### PR TITLE
🏗 Fix void tags in html-template lint rule

### PR DESCRIPTION
**Pull request by alanorozco**
Before this change, invalid void tags would be matched inconsistently. They depend on a space after the tag name `<path `, while multiline tags would be ignored due to a newline after the tagname `<path\n`.

```js
// Ignored:
html`
  <path
    d="long long long long long long long long long long long long long long long"
  />
`;

// Linted:
html`<path d="short short short" />`;
// --^
// Invalid void tag "path" (at 10:6)
```

1. We correct this so that void tags with newlines are matched.

2. This creates a problem for existing SVGs, which would previously be incorrectly ignored. Since we know the set of SVG elements that can be safely autofixed into the non-void form, we add an auto-fixer.

```js
// from:
html`<path />`;

// to:
html`<path></path>`;
```

### CODE REVIEW

File: build-system/eslint-rules/html-template.js [hunk 1]
```diff
@@ -28,7 +28,23 @@ const {
...
+function create(context) {
+  /**
+   * Tags used in <svg> elements can be safely autofixed from an invalid void
+   * tag like <path /> into its full closed form <path></path>.
+   */
+  const svgAutoFixableVoidTags = new Set([
+    'circle',
+    'ellipse',
+    'line',
+    'path',
```
Comment (erwinmombay) [inline build-system/eslint-rules/html-template.js:40]: question, is `path` for example a void tag? or would it be what is considered as "foreign elements" per https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements

Comment (alanorozco) [reply build-system/eslint-rules/html-template.js:40]: It's not a valid void tag, so I may have named the set inappropriately. The concept here is that they can be autofixed when they're self-closing.

Comment (erwinmombay) [reply build-system/eslint-rules/html-template.js:40]: yep, just clarifying the terminology since im new to the terms

Comment (alanorozco) [reply build-system/eslint-rules/html-template.js:40]: What's clear to me is that they're self-closing, so I renamed it to that.

**Review by rsimha — COMMENTED** [review]
Looks good overall, deferring to Erwin's review for the lint rule change.

**Review by erwinmombay — APPROVED** [review]

**Comment by erwinmombay** [comment]
LGTM

**Pull request merged**